tachyon/base/parallelize.h

```diff
@@ -32,6 +32,18 @@ void ParallelizeByChunkSize(Container& container, size_t chunk_size,
                             Callable callback) {
   if (chunk_size == 0) return;
   size_t num_chunks = (std::size(container) + chunk_size - 1) / chunk_size;
+  if (num_chunks == 1) {
+    SpanTy chunk(std::data(container), std::size(container));
+    if constexpr (ArgNum == 1) {
+      callback(chunk);
+    } else if constexpr (ArgNum == 2) {
+      callback(chunk, 0);
+    } else {
+      static_assert(ArgNum == 3);
+      callback(chunk, 0, chunk_size);
+    }
+    return;
+  }
```

Comment:
How about adding the following function (tested)?
```c++
template <typename Container, typename Callable,
          typename FunctorTraits = internal::MakeFunctorTraits<Callable>,
          typename RunType = typename FunctorTraits::RunType,
          typename ArgList = internal::ExtractArgs<RunType>,
          typename SpanTy = internal::GetType<0, ArgList>,
          typename T = typename SpanTy::value_type,
          size_t ArgNum = internal::GetSize<ArgList>>
void CallBack(Container& container, size_t i, size_t num_chunks,
              size_t chunk_size, Callable callback) {
  size_t len =
      i == num_chunks - 1 ? std::size(container) - i * chunk_size : chunk_size;
  SpanTy chunk(std::data(container) + i * chunk_size, len);
  if constexpr (ArgNum == 1) {
    callback(chunk);
  } else if constexpr (ArgNum == 2) {
    callback(chunk, 0);
  } else {
    static_assert(ArgNum == 3);
    callback(chunk, 0, chunk_size);
  }
}
```
then, 
```suggestion
    CallBack(container, 0, num_chunks, chunk_size, callback);
```

Reply:
Then, the code inside `OMP_PARALLEL_FOR` can also be simplified with a call the `CallBack`